models/Task.php

```diff
@@ -397,20 +413,24 @@ public function afterSave($insert, $changedAttributes)
     {
         parent::afterSave($insert, $changedAttributes);
 
-        if($this->scenario === self::SCENARIO_EDIT) {
+        if ($this->scenario === self::SCENARIO_EDIT) {
             $oldTaskUsers = $this->taskUsers;
 
             TaskUser::deleteAll(['task_id' => $this->id]);
 
+            if (true == $this->has_account) {
```

Comment:
same as `if ($this->has_account) {`

Reply:
Sure i know it's the same thing, for me i prefer the explicit one ` ($var == true) ` rather than the implicit style and i know that the first is slower than the direct check.

But specially in this case, i made that on purpose so that who sees code or who want to contribute should treat `$task->has_account` field as `boolean` and not as `integer` as it's described in the properties annotation section as `integer` for migration performance purpose : 
https://github.com/Coinsence/humhub-modules-tasks/blob/c69ff43eff1fb6f55263a6a325a22c4fabf87b58/models/Task.php#L59

Because when doing ``` if ($var) ``` $var could be integer or boolean, so that maybe someone uses $var as integer with maybe `5 as value` somewhere else in the code while `$var` is meant to be treated as boolean.

Note that i've used the Yoda-condition to avoid assignment rather than comparison.

PS : if it was a function that returns a bool i would do this `if($this->hasAccount()){ }`

Reply:
re 

> I am not sure about this exception for various reasons

I totally agree with you.

Reply:
I was missing a :) in my comment. 

thanks for the explanation, very good points!